### PR TITLE
test(common): validate that all tests are implemented

### PR DESCRIPTION
**Pull request by Haroenv**
The common test suite tests are now moved into an object of "setups" and ran separately.

This allows for a couple things that are nice
- you can easily skip running a test or run just a single one by putting a condition in the loop
- TypeScript enforces all tests are implemented
- jest enforces all tests are implemented

### CODE REVIEW

File: packages/react-instantsearch-hooks-web/src/__tests__/common-connector.test.tsx [hunk 1]
```diff
@@ -311,5 +305,25 @@ createToggleRefinementConnectorTests(
...
+        routing: true,
+      },
+    },
+  };
+
+  Object.keys(testSuites).forEach((testName) => {
+    // @ts-ignore (typescript is only referentially typed)
+    // https://github.com/microsoft/TypeScript/issues/38520
+    testSuites[testName](setups[testName], act, testOptions[testName]);
+  });
```
Comment (Haroenv) [inline packages/react-instantsearch-hooks-web/src/__tests__/common-connector.test.tsx:328]: ideally this would pass typescript, but I don't find a way yet

Comment (Haroenv) [reply packages/react-instantsearch-hooks-web/src/__tests__/common-connector.test.tsx:328]: (note that it's not related to this PR specifically, as the ts-ignore was already here on master

**Review by dhayab — APPROVED** [review]
This looks great. Do we need to explain this structure in the README of the common tests package?

**Comment by Haroenv** [comment]
the CTS readme explains only how to write the tests, which hasn't changed, so I think it's ok :)

**Pull request merged**